api/storage/storage.proto

```diff
@@ -0,0 +1,34 @@
+syntax = "proto3";
+
+package sidero.discovery.storage;
+option go_package = "github.com/siderolabs/discovery-service/api/storagepb";
+
+import "google/protobuf/timestamp.proto";
+
+// StateSnapshot is the snapshot of the discovery service state.
+//
+// We avoid marshalling/unmarshalling to this type directly, as it causes an allocation of all of the clusters at once.
```

Comment:
it feels like this comment doesn't belong here? it might be better in the Marshal/Unmarshal methods, as it doesn't change the proto definition